### PR TITLE
Use retention-days to remove non-release artifacts after 30 days

### LINKED ISSUE #2499: Workflow failure: cleanup
## Workflow Failure

[cleanup](https://github.com/pulumi/pulumi-kubernetes/blob/master/.github/workflows/artifact-cleanup.yml) has failed. See the list of failures below:

- [2023-07-16T01:33:28.000Z](https://github.com/pulumi/pulumi-kubernetes/actions/runs/5565139498)

- [2023-07-24T01:30:30.000Z](https://github.com/pulumi/pulumi-kubernetes/actions/runs/5639788271)

- [2023-07-28T01:30:00.000Z](https://github.com/pulumi/pulumi-kubernetes/actions/runs/5687177066)

- [2023-07-31T01:30:36.000Z](https://github.com/pulumi/pulumi-kubernetes/actions/runs/5709569123)

### PR DESCRIPTION
**Pull request by guineveresaenger**
Fixes #2499.

Similar to https://github.com/pulumi/pulumi-aws/pull/2649.

### CODE REVIEW

**Review by aq17 — APPROVED** [review]

File: .github/workflows/run-acceptance-tests.yml [hunk 1]
```diff
@@ -140,6 +140,7 @@ jobs:
       with:
         name: pulumi-${{ env.PROVIDER }}-provider.tar.gz
         path: ${{ github.workspace }}/bin/provider.tar.gz
+        retention-days: 30
```
Comment (mjeffryes) [inline .github/workflows/run-acceptance-tests.yml:143]: OOC: why do we upload acceptance test builds to artifactory?

**Review by mjeffryes — APPROVED** [review]
LGTM

File: .github/workflows/run-acceptance-tests.yml (hunk 1 above)
Comment (guineveresaenger) [reply .github/workflows/run-acceptance-tests.yml:143]: excellent question! we download them later in the Workflow to use for actually running the tests (~ L#215 I think).

**Comment by guineveresaenger** [comment]
@lblackstone - it sure looks suspicious, although it really shouldn't happen as a result of this change.
I will try to remove the 30 day default from the kubeconfig artifact upload and see what happens.

**Comment by guineveresaenger** [comment]
@lblackstone - the upgrade to v3 seems to have resolved the issue. I'm reasonably satisfied that otherwise there should not be any correlation between "we retain the kubeconfig generated by the PR tests for 30 days using a new functionality" and the previously failing test. WDYT?

**Pull request merged**